evalml/data_checks/sparsity_data_check.py

```diff
@@ -0,0 +1,90 @@
+from evalml.data_checks import (
+    DataCheck,
+    DataCheckMessageCode,
+    DataCheckWarning
+)
+from evalml.problem_types import ProblemTypes, handle_problem_types
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper
+)
+
+warning_too_unique = "Input columns ({}) for {} problem type are too sparse."
+
+
+class SparsityDataCheck(DataCheck):
+    """Checks if there are any columns with sparsely populated values in the input."""
+
+    def __init__(self, problem_type, threshold, unique_count_threshold=10):
+        """Checks each column in the input to determine the sparsity of the values in those columns.
+        Arguments:
+            problem_type (str or ProblemTypes): The specific problem type to data check for.
+                'multiclass' or 'time series multiclass' is the only accepted problem type.
+            threshold (float): The threshold value, or percentage of each column's unique values,
+                below which, a column exhibits sparsity.  Should be between 0 and 1.
+            unique_count_threshold (int): The minimum number of times a unique
+                value has to be present in a column to not be considered "sparse."
+        """
+        self.problem_type = handle_problem_types(problem_type)
+        if self.problem_type not in [ProblemTypes.MULTICLASS, ProblemTypes.TIME_SERIES_MULTICLASS]:
+            raise ValueError("Sparsity is only defined for multiclass problem types.")
+        self.threshold = threshold
+        if threshold < 0 or threshold > 1:
+            raise ValueError("Threshold must be a float between 0 and 1, inclusive.")
```

Comment:
Error message is different from check--we're checking between 0 and 1 exclusively :o?